src/trackColonyActions.ts

```diff
@@ -0,0 +1,92 @@
+import { getLogs } from '@colony/colony-js';
+import { utils } from 'ethers';
+
+import {
+  handleCreateDomainAction,
+  handleEditColonyAction,
+  handleEditDomainAction,
+  handleEmitDomainReputationAction,
+  handleMintTokensAction,
+  handleMoveFundsAction,
+  handlePaymentAction,
+  handleTokenUnlockedAction,
+  handleVersionUpgradeAction,
+} from '~handlers';
+import networkClient from '~networkClient';
+import { ColonyActionHandler, ContractEventsSignatures, Filter } from '~types';
+import { getCachedColonyClient, mapLogToContractEvent, verbose } from '~utils';
+
+const getFilter = (
+  eventSignature: ContractEventsSignatures,
+  colonyAddress: string,
+): Filter => ({
+  topics: [utils.id(eventSignature)],
+  address: colonyAddress,
+});
+
+const trackActionsByEvent = async (
+  eventSignature: ContractEventsSignatures,
+  colonyAddress: string,
+  handler: ColonyActionHandler,
+): Promise<void> => {
+  const colonyClient = await getCachedColonyClient(colonyAddress);
+  const filter = getFilter(eventSignature, colonyAddress);
+  const logs = await getLogs(networkClient, filter);
```

Comment:
While you can't really do anything about it in here to fix it, I think a comment needs to be left here for others who stumble on this to be aware:

This can only work with an archive node, as well as one that can display more than 10000 events. Most commercial solutions out there limit to around 10k events in the past, plus that not everyone will give up an archive node.

So this can only work with our custom, nethermind based node.